src/schema/v2/artist/targetSupply/index.ts

```diff
@@ -0,0 +1,124 @@
+import { ResolverContext } from "types/graphql"
+import { getMicrofunnelData } from "./utils/getMicrofunnelData"
+import { shuffle } from "lodash"
+
+import {
+  GraphQLObjectType,
+  GraphQLBoolean,
+  GraphQLFieldConfig,
+  GraphQLList,
+  GraphQLString,
+} from "graphql"
+import { ArtworkType } from "schema/v2/artwork"
+
+const ArtistTargetSupplyType = new GraphQLObjectType<any, ResolverContext>({
+  name: "ArtistTargetSupply",
+  fields: {
+    isInMicrofunnel: {
+      description:
+        "Returns whether an artist is in within the microfunnel list.",
+      type: GraphQLBoolean,
+      resolve: artist => Boolean(getMicrofunnelData(`/artist/${artist.id}`)),
+    },
+
+    microfunnel: {
+      type: new GraphQLObjectType<any, ResolverContext>({
+        name: "ArtistTargetSupplyMicrofunnel",
+        fields: {
+          /**
+           * This field is resolved by parsing static CSVtoJSON data.
+           * @see src/schema/v2/artist/targetSupply/utils/getMicrofunnelData.ts
+           */
+          metadata: {
+            type: new GraphQLObjectType<any, ResolverContext>({
+              name: "ArtistTargetSupplyMicrofunnelMetadata",
+              fields: {
+                highestRealized: {
+                  type: GraphQLString,
+                },
+                realized: {
+                  type: GraphQLString,
+                },
+                recentlySoldArtworkIDs: {
+                  type: new GraphQLList(GraphQLString),
+                },
+                roundedUniqueVisitors: {
+                  type: GraphQLString,
+                },
+                roundedViews: {
+                  type: GraphQLString,
+                },
+                str: {
+                  type: GraphQLString,
+                },
+                uniqueVisitors: {
+                  type: GraphQLString,
+                },
+                views: {
+                  type: GraphQLString,
+                },
+              },
+            }),
+          },
+          /**
+           * Take all of the recentlySoldArtworkIDs from metadata and perform
+           * a fetch for associated artworks and attach recently sold prices.
+           */
+          artworks: {
```

Comment:
random late question- this is a list type called `artworks`, which contains an `artwork` (somewhat redundant?), and also a price.

might this better be modeled as an artwork connection, and the edge field containing this special 'join' data (realized price). or maybe the realized price can be injected into the artwork nodes.

Reply:
The reason why it was done this way was to not pollute / complicate the Artwork type with a `realizedPrice` field, and it can't be a connection. We have a namespace (artworks) which returns artworks based on a series of specific ids (related to the `artworksByInternalID` field addition), and then we associate some metadata there.